spk-config.yaml

```diff
@@ -30,3 +30,4 @@ introspection:
     subscription_id: "subscription-id"
     tenant_id: "tenant-id"
     resource_group: "resource-group-name"
+key_vault_name: # azure key vault name"
```

Comment:
do we indicate that this is optional? should it be commented out?

Reply:
The Key Vault is optional but other parts of `spk` might make it required so I updated introspection docs to inform the user that Key Vault is optional.